nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,328 @@
+        @ReadsAttribute(attribute = "APP_KEY", description = "Specifies Azure application key for accessing the ADX-Cluster."),
+        @ReadsAttribute(attribute = "APP_TENANT", description = "Azure application tenant for accessing the ADX-Cluster."),
+        @ReadsAttribute(attribute = "CLUSTER_URL", description = "Endpoint of ADX cluster. This is required only when streaming data to ADX cluster is enabled."),
+})
+public class StandardKustoIngestService extends AbstractControllerService implements KustoIngestService {
+
+    public static final PropertyDescriptor INGEST_URL = new PropertyDescriptor
+            .Builder().name("Ingest URL")
+            .displayName("Ingest URL")
+            .description("Ingestion URL of the Azure Data Explorer cluster.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor KUSTO_AUTH_STRATEGY = new PropertyDescriptor
+            .Builder().name("Authentication Strategy")
+            .displayName("Authentication Strategy")
+            .description("Authentication method for access to Azure Data Explorer")
+            .required(false)
+            .defaultValue(KustoAuthenticationStrategy.MANAGED_IDENTITY.getValue())
+            .allowableValues(KustoAuthenticationStrategy.class)
+            .build();
+
+    public static final PropertyDescriptor APPLICATION_CLIENT_ID = new PropertyDescriptor
+            .Builder().name("Application Client ID")
+            .displayName("Application Client ID")
+            .description("Azure Data Explorer Application Client Identifier for Authentication")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor APPLICATION_KEY = new PropertyDescriptor
+            .Builder().name("Application Key")
+            .displayName("Application Key")
+            .description("Azure Data Explorer Application Key for Authentication")
+            .required(true)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(AUTHENTICATION_STRATEGY, KustoAuthenticationStrategy.APPLICATION_CREDENTIALS.getValue())
+            .build();
+
+    public static final PropertyDescriptor APPLICATION_TENANT_ID = new PropertyDescriptor.Builder()
+            .name("Application Tenant ID")
+            .displayName("Application Tenant ID")
+            .description("Azure Data Explorer Application Tenant Identifier for Authentication")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(AUTHENTICATION_STRATEGY, KustoAuthenticationStrategy.APPLICATION_CREDENTIALS.getValue())
+            .build();
+
+    public static final PropertyDescriptor CLUSTER_URI = new PropertyDescriptor
```

Comment:
Is this Cluster URI or Ingestion Cluster URI. Lets make this consistent

Reply:
Since we support managed streaming ingestion mode, we need both cluster URI and ingestion URI. rename Ingest UR URL -> Ingest URI to make it consistent with Cluster URI.

Reply:
Good point @tanmaya-panda1, I think we already have a method on the SDK where we can derive ingest- from the cluster if it does not have it and also the other way round. We may need only one URL (we can derive the other pair from it). When you get few min you can have a look at 

https://github.com/Azure/azure-kusto-java/blob/eff584c119a171f5f046dc66a3bb94881eba991d/ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientBase.java#L28C17-L28C17